scripts/build/builders/nrf.py

```diff
@@ -138,17 +141,22 @@ def generate(self):
                         'To update $ZEPHYR_BASE run: python3 scripts/setup/nrfconnect/update_ncs.py --update --shallow')
 
                     raise Exception('ZEPHYR_BASE validation failed')
-
+            if os.path.exists(os.path.join(
+                    self.root, self.app.AppPath(), "nrfconnect/boards/"+self.board.GnArgName()+".conf")):
+                boardconfig = "-DOVERLAY_CONFIG=boards/"+self.board.GnArgName()+".conf"
+            else:
+                boardconfig = ""
             cmd = '''
 source "$ZEPHYR_BASE/zephyr-env.sh";
 export GNUARMEMB_TOOLCHAIN_PATH="$PW_PIGWEED_CIPD_INSTALL_DIR";
-west build --cmake-only -d {outdir} -b {board} {sourcedir}{rpcs}
+west build --cmake-only -d {outdir} -b {board} {sourcedir} -- {boardoverlay} {rpcs}
```

Comment:
previously `--` was not needed. if boardoverlay is empty, this will result in an extra `" --  "` entry which looks ugly.

How about making the overlay config be  " -- ... " or "" (i.e. include spaces) and have the string be `... {sourcedir}{boardoverlay}{rpcs}` ? Previous code seems to do that to avoid extra spacing.

Reply:
hmm I'm surprised how the previous code could work. I tried

```
./scripts/build/build_examples.py --target-glob 'nrf*53*light-rpc' build
```

but it got this:
```
Exception: Command ['bash', '-c', 'source "$ZEPHYR_BASE/zephyr-env.sh";\nexport GNUARMEMB_TOOLCHAIN_PATH="$PW_PIGWEED_CIPD_INSTALL_DIR";\nwest build --cmake-only -d /home/lmpprk/connectedhomeip/out/nrf-nrf5340dk-light-rpc -b nrf5340dk_nrf5340_cpuapp /home/lmpprk/connect
homeip/examples/lighting-app/nrfconnect-DOVERLAY_CONFIG=rpc.overlay'] failed: 1
```

I think the " --" has to be left out of the two overlay arg variables, otherwise in case `boardoverlay` is empty and `rpcs` is not, there would be no "--" that can pass the rpc overlay flag.

How about updating to this:
```
west build --cmake-only -d {outdir} -b {board} {sourcedir} --{boardoverlay}{rpcs}
boardconfig = " -DOVERLAY_CONFIG=boards/"+self.board.GnArgName()+".conf"
rpcs=" -DOVERLAY_CONFIG=rpc.overlay"
```

Reply:
pushed a commit with the code update